<launch>

<rosparam file="$(find snoopy_setup)/all_parameters.yaml" command="load"/>

<node pkg="snoopy_setup" name="kinematics" type="kinematics" />

<node pkg="phidgets" name="left_motor" type="motor">
	<param name="serial" value="469404"/>
	<param name="name" value="left_motor"/>
</node>	

<node pkg="phidgets" name="right_motor" type="motor">
	<param name="serial" value="465530"/>
	<param name="name" value="right_motor"/>
</node>	

<node name="controller_left" pkg="snoopy_control" type="pid_controller" output="screen">
      <param name="Kp" value="1" />
      <param name="Ki" value="0.2" />
      <param name="Kd" value="0.05" />
      <param name="setpoint_topic" value="setpoint_left" />
      <param name="encoder_topic" value="left_motor/encoder" />
      <param name="control_topic" value="left_motor/cmd_vel" />
      <param name="error_topic" value="left_motor/err" />
      <param name="est_topic" value="left_motor/est" />
</node>

<node name="controller_right" pkg="snoopy_control" type="pid_controller" output="screen">
      <param name="Kp" value="1" />
      <param name="Ki" value="0.2" />
      <param name="Kd" value="0.05" />
      <param name="setpoint_topic" value="setpoint_right" />
      <param name="encoder_topic" value="right_motor/encoder" />
      <param name="control_topic" value="right_motor/cmd_vel" />
      <param name="error_topic" value="right_motor/err" />
      <param name="est_topic" value="right_motor/est" />
</node>

<node pkg="snoopy_odom" name="dead_reck" type="dead_reck"/>

<node pkg="snoopy_localize" name="pose_fusion" type="pose_fusion" output="screen"/>

<node pkg="snoopy_setup" name="tf_broadcaster" type="tf_broadcaster"/>

<node pkg="snoopy_navigate" name="nav_control_test" type="nav_control_test" output="screen" >
	<param name="position_topic" value="/snoopy/pose_fusion"/>
	<param name="angular_limit" value="2"/>
	<param name="linear_limit" value="0.2"/>
	<param name="kp_angular" value="0.5"/>
	<param name="kp_angular2" value="0.35"/>
	<param name="kp_linear" value="1"/>
</node>

<node pkg="snoopy_navigate" name="navigate_action_client" type="navigate_action_client" output="screen" />

<node pkg="rviz" name="rviz" type="rviz" />

</launch>
